src/stdlib_experimental_quadrature_simps.fypp

```diff
@@ -0,0 +1,277 @@
+        case (4)
+            integral = simps38(y, dx)
+        ! case (5) not needed; handled by default
+        case (6) ! needs special handling because of averaged 3/8's rule case
+            if (present(even)) then
+                if (even < 0) then
+                    ! 3/8 rule on left
+                    integral =  simps38(y(1:4), dx) + simps(y(4:6), dx)
+                    return
+                else if (even > 0) then
+                    ! 3/8 rule on right
+                    integral = simps(y(1:3), dx) + simps38(y(3:6), dx)
+                    return
+                else
+                    ! fall through
+                end if
+            end if
+            ! either `even` not present or is zero
+            ! equivalent to averaging left and right
+            integral = dx/48.0_${k1}$ * (17*(y(1) + y(6)) + 59*(y(2) + y(5)) + 44*(y(3) + y(4)))
+        case default
+            if (mod(n, 2) == 1) then
+                integral = dx/3.0_${k1}$*(y(1) + 4*sum(y(2:n-1:2)) + 2*sum(y(3:n-2:2)) + y(n))
+            else
+                if (present(even)) then
+                    if (even < 0) then
+                        ! 3/8th rule on left
+                        integral = simps38(y(1:4), dx) + simps(y(4:n), dx)
+                        return
+                    else if (even > 0) then
+                        ! 3/8 rule on right
+                        integral = simps(y(1:n-3), dx) + simps38(y(n-3:n), dx)
+                        return
+                    else
+                        ! fall through
+                    end if
+                end if
+                ! either `even` not present or is zero
+                ! equivalent to averaging left and right
+                integral = dx/48.0_${k1}$ * (17*(y(1) + y(n)) + 59*(y(2) + y(n-1)) &
+                        + 43*(y(3) + y(n-2)) + 49*(y(4) + y(n-3)) + 48*sum(y(5:n-4)))
+            end if
+        end select
+    end function simps_dx_${k1}$
+
+#:endfor
+#:for k1, t1 in REAL_KINDS_TYPES
+
+    pure recursive module function simps_x_${k1}$(y, x, even) result(integral)
+        ${t1}$, dimension(:), intent(in) :: y
+        ${t1}$, dimension(size(y)), intent(in) :: x
+        integer, intent(in), optional :: even
+        ${t1}$ :: integral
+
+        integer :: i
+        integer :: n
+
+        ${t1}$ :: h1, h2, h3
+        ${t1}$ :: a, b, c, d
+
+        n = size(y)
+
+        select case (n)
+        case (0:1)
+            integral = 0.0_${k1}$
+        case (2)
+            integral = 0.5_${k1}$*(x(2) - x(1))*(y(1) + y(2))
+        case (3)
+            h1 = x(2) - x(1)
+            h2 = x(3) - x(2)
+            a = (2*h1**2 + h1*h2 - h2**2)/(6*h1)
+            b = (h1+h2)**3/(6*h1*h2)
+            c = (2*h2**2 + h1*h2 - h1**2)/(6*h2)
+            integral = a*y(1) + b*y(2) + c*y(3)
+        case (4)
+            integral = simps38(y, x)
+        ! case (6) unneeded?
+        case default
```

Comment:
Here too: all arrays with a size >4 will end up in the `case default`. Is it intended?

Reply:
Yep, same comment as above. Though, the questioning "unneeded?" can be changed to a firm "unneeded."